politeiawww/backend.go

```diff
@@ -2277,8 +2277,16 @@ func (b *backend) ProcessLikeComment(lc www.LikeComment, user *database.User) (*
 		}
 	}
 
+	// Comment exists?
+	c, err := b._getInventoryRecordComment(lc.Token, lc.CommentID)
```

Comment:
`b._getInventoryRecordComment` must be called with the mutex locked.  You're creating a race condition by calling it without locking the mutex.  

You don't need to call `b._getInventoryRecordComment` since we already lookup the inventory record right here:
https://github.com/decred/politeia/pull/595/files#diff-d9012ef76f0533a087446113b19c820eR2297

The inventory record contains all the comments for the proposal, so you can use `lc.CommentID` to make sure that the passed in comment exists.

Reply:
Hey @lukebp, I appreciate the review. I had missunterstood how I'd access the inventory, but it's clear to me now, thanks!

Changed it and now I check the comment existence on the loaded record `ir` from the inventory.